src/css_composer/view/CssRulesView.js

```diff
@@ -61,19 +62,41 @@ module.exports = require('backbone').View.extend({
       rendered = view.render().el;
     }
 
+    const mediaWidth = this.getMediaWidth(model.get('mediaText'));
+    const styleBlockId = `${this.pfx}rules-${mediaWidth}`;
+
     if (rendered) {
-      if (fragment) fragment.appendChild(rendered);
-      else this.$el.append(rendered);
+      if (fragment) {
+        fragment.getElementById(styleBlockId).appendChild(rendered);
+      } else {
+        let $stylesContainer = this.$el.find(`#${styleBlockId}`);
+        $stylesContainer.append(rendered);
+      }
     }
 
     return rendered;
   },
 
+  getMediaWidth(mediaText) {
+    return mediaText && mediaText.replace('(max-width: ', '').replace(')', '');
```

Comment:
`(max-width: ` it's not always the same, it depends on [`mediaCondition ` option](https://github.com/artf/grapesjs/blob/dev/src/editor/config/config.js#L91)

Reply:
Can I somehow simplify it? Do I have an option to take the current media condition parameter from the model directly - without any replacements?

Reply:
You should be able to get it directly from the config `this.em.getConfig(). mediaCondition`